egs/mobvoi/v1/local/chain/tuning/run_tdnn_1a.sh

```diff
@@ -0,0 +1,293 @@
+  
+  prefinal-layer name=prefinal-chain input=prefinal-l $prefinal_opts big-dim=$dim small-dim=30
+  output-layer name=output include-log-softmax=false dim=$num_targets $output_opts
+
+  prefinal-layer name=prefinal-xent input=prefinal-l $prefinal_opts big-dim=$dim small-dim=30
+  output-layer name=output-xent dim=$num_targets learning-rate-factor=$learning_rate_factor $output_opts
+EOF
+  steps/nnet3/xconfig_to_configs.py --xconfig-file $dir/configs/network.xconfig --config-dir $dir/configs
+fi
+
+if [ $stage -le 6 ]; then
+  # no need to store the egs in a shared storage because we always
+  # remove them. Anyway, it takes only 5 minutes to generate them.
+
+  cp $tree_dir/phone_lm.fst $dir/phone_lm.fst
+
+  steps/nnet3/chain/train.py --stage=$train_stage \
+    --cmd="$decode_cmd" \
+    --feat.cmvn-opts="--config=conf/online_cmvn.conf" \
+    --chain.xent-regularize $xent_regularize \
+    --chain.leaky-hmm-coefficient=0.1 \
+    --chain.l2-regularize=0.0 \
+    --chain.apply-deriv-weights=false \
+    --trainer.add-option="--optimization.memory-compression-level=2" \
+    --trainer.srand=$srand \
+    --trainer.max-param-change=2.0 \
+    --trainer.num-epochs=$num_epochs \
+    --trainer.frames-per-iter $frames_per_iter \
+    --trainer.optimization.num-jobs-initial $num_jobs_initial \
+    --trainer.optimization.num-jobs-final $num_jobs_final \
+    --trainer.optimization.initial-effective-lrate 0.00005 \
+    --trainer.optimization.final-effective-lrate 0.000005 \
+    --trainer.optimization.backstitch-training-scale $bs_scale \
+    --trainer.num-chunk-per-minibatch=128,64 \
+    --trainer.optimization.momentum=0.0 \
+    --egs.chunk-width=$chunk_width \
+    --egs.chunk-left-context=0 \
+    --egs.chunk-right-context=0 \
+    --egs.chunk-left-context-initial=0 \
+    --egs.chunk-right-context-final=0 \
+    --egs.dir="$common_egs_dir" \
+    --egs.opts="--frames-overlap-per-eg 0 --online-cmvn $online_cmvn" \
+    --cleanup.remove-egs=$remove_egs \
+    --use-gpu=true \
+    --reporting.email="$reporting_email" \
+    --feat-dir $combined_train_data_dir \
+    --tree-dir $tree_dir \
+    --lat-dir=$combined_lat_dir \
+    --dir=$dir  || exit 1;
+fi
+
+if [ $stage -le 7 ]; then
+  steps/online/nnet3/prepare_online_decoding.sh \
+    --mfcc-config conf/mfcc_hires.conf \
+    --online-cmvn-config conf/online_cmvn.conf \
+    $lang ${dir} ${dir}_online
+
+  rm $dir/.error 2>/dev/null || true
+  for wake_word_cost in -0.5 0.0 0.5 1.0 1.5 2.0 2.5 3.0 3.5 4.0 4.5 5.0; do
+    rm -rf $lang_decode
+    utils/prepare_lang.sh --num-sil-states 1 --num-nonsil-states 4 --sil-prob 0.0 \
+      --position-dependent-phones false \
+      data/local/dict "<sil>" $lang_decode/temp $lang_decode
+
+    sil_id=`cat $lang_decode/words.txt | grep "<sil>" | awk '{print $2}'`
+    freetext_id=`cat $lang_decode/words.txt | grep "FREETEXT" | awk '{print $2}'`
+    id=`cat $lang_decode/words.txt | grep $wake_word | awk '{print $2}'`
```

Comment:
```bash
lc_all=$LC_ALL
export LC_ALL=en_US.UTF-8
id=`cat $lang_decode/words.txt | grep $wake_word | awk '{print $2}'`
export LC_ALL=$lc_all
```

Otherwise, `id_word` is `file`, not 2. And it will fail in the subsequent fst compilation.

Reply:
I don't have such problem